src/app/settings/views/Security/SessionTimeout/SessionTimeout.tsx

```diff
@@ -0,0 +1,82 @@
+import { Spinner } from "@canonical/react-components";
+import { formatDuration } from "date-fns";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { useWindowTitle } from "app/base/hooks";
+import { actions as configActions } from "app/store/config";
+import configSelectors from "app/store/config/selectors";
+import { humanReadableToSeconds, secondsToDuration } from "app/utils/timeSpan";
+
+type SessionTimeoutFormValues = {
+  session_length: string;
+};
+
+export enum Labels {
+  Loading = "Loading...",
+  Expiration = "Session timeout expiration",
+  Save = "Save",
+  ConfigureSessionTimeout = "Configure Session Timeout",
+}
+
+const SessionTimeoutSchema = Yup.object().shape({
+  session_length: Yup.string()
+    .required("Timeout length is required")
+    .matches(
+      /^((\d)+ ?(hours|days|weeks|months|years) ?(and)? ?)+$/,
+      "Unit must be `string` type with a value of years, months, weeks, days, and/or hours."
+    ),
```

Comment:
It might be better to use a Yup test function using `human-interval` to assert if the value is valid instead as the library handles singular and plural values.
https://github.com/jquense/yup#schematestname-string-message-string--function--any-test-function-schema

Reply:
I can do that, I think the only issue is that we don't allow minutes and seconds as time units so it might still require some regex trickery

Reply:
```suggestion
    .matches(
      /^((\d)+ ?(hour|day|week|month|year)(s)? ?(and)? ?)+$/,
      "Unit must be `string` type with a value of years, months, weeks, days, and/or hours."
    ),
```

This can also be fixed by doing the above

Reply:
Ah right, that's fine. Let's update the regex in that case.